### PR TITLE
extension of #3983

### PR DESCRIPTION
**Pull request by Sentient07**
gh-3983

### CODE REVIEW

File: theano/tensor/xlogx.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
+from __future__ import division
 import numpy
-
```
Comment (nouiz) [inline theano/tensor/xlogx.py]: keep the empty line. Pep8 tell us that we need to separate the import from out project of the others. In fact, there should be 3 import section import from python default library, import of 3th party python library, import from this library.

Comment (Sentient07) [inline theano/tensor/xlogx.py]: @nouiz Made that change and will keep this in mind. :+1:

File: theano/scalar/basic.py [hunk 2]
```diff
@@ -604,12 +604,11 @@ def __sub__(self, other):
     def __mul__(self, other):
         return mul(self, other)
 
-    if PY3:
-        def __truediv__(self, other):
-            return div_proxy(self, other)
-    else:
-        def __div__(self, other):
-            return div_proxy(self, other)
```
Comment (lamblin) [inline theano/scalar/basic.py]: @nouiz Is that actually correct? What happens if someone calls `a / b` on int tensors, from outside of Theano, without having imported `division` from `__future__`?

Comment (Sentient07) [inline theano/scalar/basic.py]: @lamblin @nouiz , I had tried this from outside theano without importing ```__future__``` , 

`>>> c = theano.function([a,b], b/a)
>>> c(2,5)
array(2, dtype=int32)
>>> c = theano.function([a,b], b//a)
>>> c(3,5)
array(1, dtype=int32)
>>> c(3,6)
array(2, dtype=int32)
>>> c = theano.function([a,b], b/a)
>>> c(3,6)
array(2, dtype=int32)
`

Comment (nouiz) [inline theano/scalar/basic.py]: in tensor, \__div__ is always div_proxy. But \__truediv__ is always true_div. There is code that make truediv be div, but it is overrided later.
Without the futur import.
python 2, int / int give and int // int give int.
Python 3, int / int give float and int // int give int.
numpy 1.10 behave like python 2.

With futur import,
python 2 do as python 3
numpy int also do as python 3.

div_proxy always return int_div by default. This is a theano flag operation.

I think we should do as numpy 1.10, we do as python 2/3 depending of the version used and if futur is imported, we behave like python 3. 

It is possible that older version of numpy do differently.

Can you test that with this code in python 2, in a file that just import theano, we keep the python 2 semantic, but in a file where the futur import is there, we use the python 3 semantic?

Comment (Sentient07) [inline theano/scalar/basic.py]: @nouiz, I've to test ```//``` and ```/``` on a tensor with and without future in python2 and test if the one with future behaves as python3 ?

Comment (nouiz) [inline theano/scalar/basic.py]: I think we need to add 2 tests file:
- theano/scalar/tests/test_div_no_futur.py
- theano/scalar/tests/test_div_futur.py

They should be mostly the same, except that one do the import about futur. They should check that we do the correct choise for the give python that is currently running.

This way, we will be sure we do it correctly and that it don't get broken in the futur.

Comment (Sentient07) [inline theano/scalar/basic.py]: Yeah, sure. Will add the tests

Comment (Sentient07) [inline theano/scalar/basic.py]: @nouiz Added the tests.

**Comment by Sentient07** [comment]
@nouiz ready for merge ? I'll fix those merge conflicts in #3983 by tonight

**Comment by Sentient07** [comment]
@lamblin ping. I guess this is ready for merge ?

**Comment by lamblin** [comment]
Sure, thanks.

**Pull request merged**